sips/sip-013/sip-013-semi-fungible-token-standard.md

```diff
@@ -0,0 +1,388 @@
+
+## Trait definition
+
+A definition of the trait is provided below.
+
+```clarity
+(define-trait sip013-semi-fungible-token-trait
+	(
+		;; Get a token type balance of the passed principal.
+		(get-balance (uint principal) (response uint uint))
+
+		;; Get the total SFT balance of the passed principal.
+		(get-overall-balance (principal) (response uint uint))
+
+		;; Get the current total supply of a token type.
+		(get-total-supply (uint) (response uint uint))
+
+		;; Get the overall SFT supply.
+		(get-overall-supply () (response uint uint))
+
+		;; Get the number of decimal places of a token type.
+		(get-decimals (uint) (response uint uint))
+
+		;; Get an optional token URI that represents metadata for a specific token.
+		(get-token-uri (uint) (response (optional (string-ascii 256)) uint))
+
+		;; Transfer from one principal to another.
+		(transfer (uint uint principal principal) (response bool uint))
+
+		;; Transfer from one principal to another with a memo.
+		(transfer-memo (uint uint principal principal (buff 34)) (response bool uint))
+	)
+)
+```
+## Events
+
+Semi-fungible token contracts should emit custom events in certain situations
+via `print`. These events should be emitted after any built-in token events
+(such as those emitted by `ft-transfer?`) and before the memo in the case of
+`transfer-memo` and `transfer-many-memo`.
+
+| Event name           | Tuple structure                                                                                 | Description                          |
+|----------------------|-------------------------------------------------------------------------------------------------|--------------------------------------|
+| `sft_transfer`       | `{type: "sft_transfer", token-id: uint, amount: uint, sender: principal, recipient: principal}` | Emitted when tokens are transferred. |
+| `sft_mint`           | `{type: "sft_mint", token-id: uint, amount: uint, recipient: principal}`                        | Emitted when new tokens are minted.  |
+| `sft_burn`           | `{type: "sft_burn", token-id: uint, amount: uint, sender: principal}`                           | Emitted when tokens are burned.      |
+
+## Use of native asset functions
+
+Contract implementers should always use the built-in native assets that are
+provided as Clarity primitives whenever possible. This allows clients to use
+Post Conditions (explained below) and takes advantage of other benefits like
+native events and asset balances. However, there are no language primitives
+specific to semi-fungible tokens. The reference implementation included in this
+SIP therefore leverages the primitives to the extent that Clarity allows for.
+
+The recommended native asset primitives to use:
+
+- `define-fungible-token`
+- `ft-burn?`
+- `ft-get-balance`
+- `ft-get-supply`
+- `ft-mint?`
+- `ft-transfer?`
+- `define-non-fungible-token`
+- `nft-burn?`
+- `nft-mint?`
+- `nft-transfer?`
```

Comment:
Can you add a minimal example to the text here for using the `ft-transfer?` and `nft-transfer?` methods to implement a `transfer` method?  I know that the reference implementations will include this, but I think a minimal example inline will help readers see how they can use native asset functions to preserve post-condition functionality.

Reply:
@MarvinJanssen  I think you have control of the fork - am not sure how to push a commit to this PR?

Reply:
@kantai just pushed an update. Is it what you are looking for?

Reply:
As mentioned, I'm filling in for @kantai here. This looks good to me.